### PR TITLE
Query available models of openrouter.ai provider

### PR DESCRIPTION
**Pull request by zimmski**

### CODE REVIEW

File: provider/openrouter/openrouter.go [hunk 1]
```diff
@@ -29,11 +29,17 @@ func newOpenRouterProvider() provider.Provider {
...
 	config := openai.DefaultConfig(o.token)
 	config.BaseURL = o.baseURL
-	client := openai.NewClientWithConfig(config)
+
+	return openai.NewClientWithConfig(config)
+}
+
+// Query queries the LLM.
+func (o *openRouterProvider) Query(ctx context.Context, modelIdentifier string, promptText string) (response string, err error) {
+	client := o.client()
```
Comment (bauersimon) [inline provider/openrouter/openrouter.go]: No need to instantiated the client even... we could do `o.client().CreateChatCompletion...` directly.

File: provider/openrouter/openrouter.go [hunk 2]
```diff
@@ -74,21 +74,17 @@ func (p *openRouterProvider) ID() (id string) {
...
-	modelIDs := []string{
-		"google/gemma-7b-it:free",
-		"gryphe/mythomist-7b:free",
-		"mistralai/mistral-7b-instruct:free",
-		"nousresearch/nous-capybara-7b:free",
-		"openrouter/cinematika-7b:free",
-		"undi95/toppy-m-7b:free",
+func (p *openRouterProvider) Models() (models []model.Model, err error) {
+	client := p.client()
+	ms, err := client.ListModels(context.Background())
```
Comment (bauersimon) [inline provider/openrouter/openrouter.go:79]: No need to instantiated the client even... we could do `o.client().CreateChatCompletion...` directly.

File: provider/openrouter/openrouter.go (hunk 1 above)
Comment (zimmski) [reply provider/openrouter/openrouter.go]: Prefer a client object for a clean configuration.

File: provider/openrouter/openrouter.go (hunk 2 above)
Comment (zimmski) [reply provider/openrouter/openrouter.go:79]: Same as https://github.com/symflower/eval-symflower-codegen-testing/pull/12#discussion_r1549576437

**Pull request merged**